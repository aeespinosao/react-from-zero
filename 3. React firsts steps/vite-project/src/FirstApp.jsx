import PropTypes from 'prop-types';

export const FirstApp = ({title = "Hi, how you doing?", subTitle}) => {

    return (
        <>
            <h1>{title}</h1>
            <p>{subTitle}</p>
        </>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.number
}

FirstApp.defaultProps = {
    subTitle: 'No hay subtitulo'
}