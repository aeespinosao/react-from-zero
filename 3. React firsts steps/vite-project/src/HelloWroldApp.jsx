/*export function App() {
    return (<h1>Hello world!!</h1>);
}

export default App;*/

export const HelloWroldApp = () => {
    return (<h1>Hello world!!</h1>);
}
