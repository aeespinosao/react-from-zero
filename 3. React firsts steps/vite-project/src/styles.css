html, body {
    background-color: cadetblue;
    color: white;
    font-size: 1.3rem;
}

button {
    padding: 15px;
    font-size: 1.3rem;  
}