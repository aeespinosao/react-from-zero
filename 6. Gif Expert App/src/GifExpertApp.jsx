import { useState } from "react";
import { AddCategory } from "./components/AddCategory";

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(["test"]);

    const onAddCategory = () => {
        setCategories([...categories, 'valorant'])
    }

    return (
        <>
            <h1>GifExpertApp</h1>

            <AddCategory />
            <button onClick={onAddCategory}>Agregar</button>
            <ol>
                { 
                    categories.map(category => { return <li key={ category }>{category}</li> }) 
                }
            </ol>
        </>
    )
}