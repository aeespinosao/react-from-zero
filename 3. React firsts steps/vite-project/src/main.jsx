import React from "react";
import ReactDOM from "react-dom/client";
import { HelloWroldApp } from "./HelloWroldApp";

ReactDOM.createRoot(document.getElementById("root")).render(
    <React.StrictMode>
        <HelloWroldApp />
    </React.StrictMode>
)