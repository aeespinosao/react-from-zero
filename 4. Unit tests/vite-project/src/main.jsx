import React from "react";
import ReactDOM from "react-dom/client";
import { CounterApp } from "./CounterApp";
import { FirstApp } from "./FirstApp";
import { HelloWroldApp } from "./HelloWroldApp";

import "./styles.css";


ReactDOM.createRoot(document.getElementById("root")).render(
    <React.StrictMode>
        <FirstApp value={21}/>
    </React.StrictMode>
)