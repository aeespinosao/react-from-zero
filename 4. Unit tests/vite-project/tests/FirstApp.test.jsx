import { render } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('Test FirstApp componen', () => {
    test('match with snapshot', () => {
        const title = 'Hola, soy Goku!!';
        const {container} = render(<FirstApp title={title}/>);
        expect(container).toMatchSnapshot();
    })

    test('title should be h1 tag', () => {
        const title = 'Hola, soy Goku!!';
        const {container, getByText} = render(<FirstApp title={title}/>);
        expect(getByText(title)).toBeTruthy();

        const h1 = container.querySelector('h1');
        expect(h1.innerHTML).toContain(title);
    })
})