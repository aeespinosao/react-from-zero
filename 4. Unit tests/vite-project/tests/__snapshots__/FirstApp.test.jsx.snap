// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FirstApp componen match with snapshot 1`] = `
<div>
  <h1>
    Hola, soy Goku!!
  </h1>
  <p>
    No hay subtitulo
  </p>
</div>
`;
